File: scripts/moxi/embed-chunks.ts
import fs from "fs/promises";
import { OpenAI } from "openai";          // points at Ollama locally

const api = new OpenAI({ baseURL: "http://localhost:11434" });

const chunks: string[] = JSON.parse(await fs.readFile("chunks.json", "utf8"));
const embeds: [string, number[]][] = [];

for (const c of chunks) {
  const e = (await api.embeddings.create({
    model: "all-minilm",                 // pulled with `ollama pull all-minilm`
    input: c
  })).data[0].embedding;
  embeds.push([c, e]);
}
await fs.writeFile("embeds.tsv",
  embeds.map(([t, v]) => `${t.replace(/\n/g," ")}\t${JSON.stringify(v)}`).join("\n"));

File: scripts/moxi/embed-chunks.py
#!/usr/bin/env python3
import json, pathlib, os, pgvector.psycopg
from psycopg import connect
from ollama import Client

DB_URL = os.getenv("DATABASE_URL", "postgresql://localhost/podcasts")
MODEL_EMB = "all-minilm"          # already `ollama pull`-ed
CHUNKS = pathlib.Path("chunks.json").read_text()
chunks = json.loads(CHUNKS)

conn = connect(DB_URL, autocommit=True)
conn.execute("CREATE EXTENSION IF NOT EXISTS vector")
conn.execute("""
  CREATE TABLE IF NOT EXISTS pod_chunks(
    id serial PRIMARY KEY,
    text text,
    emb vector(384)
  )""")

cli = Client(host="http://localhost:11434")

for c in chunks:
  v = cli.embeddings(model=MODEL_EMB, prompt=c)["embedding"]
  conn.execute("INSERT INTO pod_chunks(text, emb) VALUES (%s, %s)", (c, v))

print(f"✔  stored {len(chunks)} chunks")
